src/compressed_tensors/utils/match.py

```diff
@@ -325,7 +325,7 @@ def match_modules_set(
                 matches[target].append(module)
                 parent_context = new_parent_context
                 unmatched_targets -= {target}
-                matched_targets_for_cur_module += {target}
```

Comment:
I feel like you should be using `.add` here (instead of creating a new one element set and unioning them) but this is fine too.